src/universal/containers/Action/ActionContainer.js

```diff
@@ -31,25 +23,17 @@ export default class ActionContainer extends Component {
     params: PropTypes.object
   };
 
+  getChildContext() {
+    return {previousLocation};
+  }
+
   componentWillMount() {
-    const {dispatch} = this.context.store;
-    const {location: {pathname: nextPage}} = this.props;
-    updateAnalyticsPage(dispatch, '', nextPage);
     injectGlobals(globalStyles);
   }
 
-  componentDidUpdate(prevProps) {
-    const {location: {pathname: lastPage}} = prevProps;
-    const {location: {pathname: nextPage}} = this.props;
-    if (lastPage !== nextPage) {
-      const {dispatch} = this.context.store;
-      /*
-       * Perform page update after component renders. That way,
-       * document.title will be current after any child <Helmet />
-       * element(s) are rendered.
-       */
-      updateAnalyticsPage(dispatch, lastPage, nextPage);
-    }
+  componentWillReceiveProps() {
```

Comment:
Ah, I see:

> finally, this creates a previousLocation singleton that is passed in as context in ActionContainer. This is necessary because segment wants to know the referrer AND the params. Params are not established until it hits the bottom of the routing tree. sending them back to the top a la redux would not work because that would cause a new render cycle, which means lastPage == nextPage. So, sending down the lastPage (now called lastPath since it is a pathname) is necessary.